### PR TITLE
[s] Fixes an exploit in chef order console that potentially let players lag the server

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
If a player spoofs the amount of a specific item they want to buy, for example potatoes. When calculating the cart's total get_total_cost would loop through every individual number in the amount that they buy, grab the price of the item, then add it to a total. It seems like this causes the server to freeze as it goes through the loop if the amount is "infinite", I'm not completely sure though.

## Why It's Good For The Game
I have a pretty decent desktop rig, and it appears to slow the server to a crawl while testing. For obvious reasons, I couldn't test this on the live server.

## Changelog

:cl:
fix: Fixes an exploit in chef order console that let players potentially lag the server.
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/chef_orders/chef_order.dm [hunk 1]
```diff
@@ -33,8 +33,7 @@
 /obj/machinery/computer/chef_order/proc/get_total_cost()
 	. = 0
 	for(var/datum/orderable_item/item as anything in grocery_list)
-		for(var/i in 1 to grocery_list[item]) //for how many times we bought it
-			. += item.cost_per_order //add its price
+		. += (grocery_list[item] * item.cost_per_order)
```
Comment (LemonInTheDark) [inline code/game/machinery/computer/chef_orders/chef_order.dm:36]: you don't need the ()'s here, s just fine without em

File: code/game/machinery/computer/chef_orders/chef_order.dm [hunk 2]
```diff
@@ -71,7 +70,7 @@
 	var/datum/orderable_item/wanted_item = locate(params["target"]) in order_datums
 	switch(action)
 		if("cart_set")
-			grocery_list[wanted_item] = params["amt"]
+			grocery_list[wanted_item] = clamp(params["amt"], 0, 50)
```
Comment (LemonInTheDark) [inline code/game/machinery/computer/chef_orders/chef_order.dm:73]: Why 50?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
lmao

File: code/game/machinery/computer/chef_orders/chef_order.dm (hunk 2 above)
Comment (CocaColaTastesGood) [reply code/game/machinery/computer/chef_orders/chef_order.dm:73]: Should be 20, but the javascript file seems to hint that it can be as high as 20.

**Pull request merged**